src/backend/distributed/executor/multi_utility.c

```diff
@@ -1661,23 +1687,55 @@ ProcessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand)
  * statement needs distributed execution but contains unsupported options.
  */
 static bool
-IsSupportedDistributedVacuumStmt(Oid relationId, VacuumStmt *vacuumStmt)
+IsSupportedDistributedVacuumStmt(VacuumStmt *vacuumStmt, List *vacuumRelationIdList)
 {
 	const char *stmtName = (vacuumStmt->options & VACOPT_VACUUM) ? "VACUUM" : "ANALYZE";
 	bool distributeStmt = false;
+	ListCell *relationIdCell = NULL;
+	int distributedRelationCount = 0;
```

Comment:
TBH I'd just assign the `list_length` to `distributedRelationCount` right here and check for zero in the first `if` statement (with a suitable comment explaining `0` means _all relations_.

Reply:
ok